### ISSUE TITLE
Update middleman-core runtime dependancy to use with middleman v4

### ISSUE THREAD

**Issue by maximilianmessing** [Aupajo/middleman-search_engine_sitemap#6]
The dependancies are out of date. The middleman-core runtime dependancy should be bumped to at least 4.0.

**Comment by Aupajo** [comment]
Thanks, @maximilianmessing. I haven't tested this against 4.0 yet.

**Comment by angeloashmore** [comment]
The Padrino website ([padrino/padrino-web-new][0]) just got updated to Middleman 4 and used this gem. I'll spend some time today to see if I can get everything working. :smile:

[0]:http://github.com/padrino/padrino-web-new

**Comment by angeloashmore** [comment]
See PR #7

**Comment by maximilianmessing** [comment]
:+1:

**Comment by Aupajo** [comment]
Thanks for raising the issue. Middleman 4 is now supported. You can update your project by altering your `Gemfile` and running `bundle`:

```ruby
gem 'middleman-search_engine_sitemap', '~> 1.4.0'
```

Let me know if you run into any issues with the new version!

**Comment by maximilianmessing** [comment]
perfect! Thank you

**Comment by komor72** [comment]
Do I correctly understand, that to use `middleman-search_engine_sitemap` with last Middleman v3 version (3.4.1 as of now) I need to use 1.3.0 version of this gem?

```
gem 'middleman-search_engine_sitemap', , '1.3.0'
```

**Comment by Aupajo** [comment]
@komor72 Try it with `1.4.0` first, and if that doesn't work for you, you can use `1.3.x`:

```ruby
gem 'middleman-search_engine_sitemap', '~> 1.3.0'

**Comment by komor72** [comment]
`gem update middleman-search_engine_sitemap` caused a few of gem updates (padrino-support, padrino-helpers) and a hell of new installs (fast_blank, parallel, servolux, dotenv, addressable, memoist, fastimage, contracts, hashie, concurrent-ruby, hamster, backports), but first of all `middleman-core` was bumped to 4.1.10, which I think is not a good idea to use with Middleman v3. 
Since so many new gems have been installed (new dependencies of middleman-core 4.x?), I’ll revert to `middleman-search_engine_sitemap` v1.3 and stick with it. Maybe it’s a good idea to put this to documentation: last version for `middleman` v3.x is `middleman-search_engine_sitemap` v1.3?

**Comment by komor72** [comment]
Just out of curiosity I tried `bundle install` after aforementioned `gem update middleman-search_engine_sitemap ` but, as expected, it doesn’t go through:

```
Bundler could not find compatible versions for gem "middleman-core":
  In snapshot (Gemfile.lock):
    middleman-core (= 3.4.1)

  In Gemfile:
    middleman (= 3.4.1) was resolved to 3.4.1, which depends on
      middleman-core (= 3.4.1)
(…)
    middleman-search_engine_sitemap (~> 1.4.0) was resolved to 1.4.0, which depends on
      middleman-core (~> 4.0)
```

**Comment by Aupajo** [comment]
Thanks, @komor72. I think that's right. I rely on some private methods inside Middleman that aren't backwards-compatible to make this work. I'll update the README.
